docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,221 @@
+introduction an investment to anticipate other needs.
+
+Since the RAG behavior is the only functionality of this new command, executions without the `--rag` option will result
+in no output for now.
+
+#### Command Purpose
+Generate the embeddings from the documents at */path/to/docs/folder* folder and store them in the
+configured vector database. These embeddings are intended to be used as augmented context in a RAG-based chat pipeline.
+
+#### Assumptions
+The provided documents must be in JSON format according to the InstructLab schema: this is the schema generated
+when transforming knowledge documents with the `ilab data generate` command (see 
+[Getting Started with Knowledge Contributions][1]).
+
+To simplify the execution of the transformation step, we introduce a `--transform` option which also includes the documents
+transformation, leveraging on the `instructlab-sdg` modules. 
+
+### Why We Need It
+This command streamlines the `ilab data generate` pipeline and eliminates the requirement to define a `qna` document,
+which typically includes:
+* A minimum of 5×3 question-and-answer pairs.
+* Reference documents stored in Git.
+
+The goal is not to generate training data for InstructLab-trained models but to utilize the documents for RAG 
+workflows with pre-tuned models.
+
+#### Supported Databases
+The command may support various vector database types. A default configuration will align with the selected 
+InstructLab technology stack.
+
+#### Usage
+The generated embeddings can later be retrieved to enrich the context for RAG-based chat pipelines.
+
+#### Defining Command Options
+To maintain compatibility and simplicity, no new configurations will be introduced for this command. Instead,
+the settings will be defined using the following hierarchy (options higher in the list overriding those below):
+* CLI flags (e.g., `--rag`).
+* Environment variables following a consistent naming convention, such as `ILAB_<UPPERCASE_ARGUMENT_NAME>`.
+* Default values, for all the applicable use cases.
+
+For example, the `splitter_split_by` argument can be implemented using the `click` module like this:
+```py
+@click.option(
+    "--splitter-split-by",
+    default='word',
+    envvar="ILAB_SPLITTER_SPLIT_BY",
+)
+```
+
+### 2.3 Embedding Ingestion Pipeline Options
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Whether to include a transformation step. | `False` | `--transform` (boolean) | `ILAB_TRANSFORM` |
+| The output path of transformed documents (serve as input for the embedding ingestion pipeline). Mandatory when `--transform` is used. |  | `--transform-output` | `ILAB_TRANSFORM_OUTPUT` |
+| How to split the documents. One of `page`, `passage`, `sentence`, `word`, `line` | `word` | `--splitter-split-by` | `ILAB_SPLITTER_SPLIT_BY` |
```

Comment:
Has there been discussion about the existence of this logic with respect to the docling-based document transformation?

Reply:
You may have seen on Slack that the Docling hybrid chunker is now released.  I looked at the code briefly, and it looks good to me.  More details:
- [Announcement](https://github.com/DS4SD/docling/discussions/548)
- [Documentation](https://ds4sd.github.io/docling/examples/hybrid_chunking/)
- [Code](https://github.com/DS4SD/docling-core/blob/main/docling_core/transforms/chunker/hybrid_chunker.py)

I think this will be a good fit for the RAG chunking because (unlike their older hierarchical chunker), it provides chunks that are constrained to be no bigger than a fixed size for a given tokenizer and tries to make the chunks as big as possible within that size limit and the constraints of the structure.

Reply:
I will sync up with ET engineers to adopt the same approach once available.